ng2-components/ng2-activiti-form/src/components/form-field/form-field.component.ts

```diff
@@ -70,25 +70,26 @@ export class FormFieldComponent implements OnInit, OnDestroy {
     }
 
     ngOnInit() {
-        if (this.field) {
-            let customTemplate = this.field.form.customFieldTemplates[this.field.type];
-            if (customTemplate && this.hasController(this.field.type)) {
-                let factory = this.getComponentFactorySync(this.field.type, customTemplate);
+        let originalField = this.getField();
+        if (originalField) {
+            let customTemplate = this.field.form.customFieldTemplates[originalField.type];
+            if (customTemplate && this.hasController(originalField.type)) {
+                let factory = this.getComponentFactorySync(originalField.type, customTemplate);
                 this.componentRef = this.container.createComponent(factory);
                 let instance: any = this.componentRef.instance;
                 if (instance) {
-                    instance.field = this.field;
+                    instance.field = originalField;
                 }
             } else {
-                let componentType = this.formRenderingService.resolveComponentType(this.field);
+                let componentType = this.formRenderingService.resolveComponentType(originalField);
                 if (componentType) {
                     let factory = this.componentFactoryResolver.resolveComponentFactory(componentType);
                     this.componentRef = this.container.createComponent(factory);
                     let instance = <WidgetComponent> this.componentRef.instance;
                     instance.field = this.field;
                     instance.fieldChanged.subscribe(field => {
-                        if (field && field.form) {
-                            this.visibilityService.refreshVisibility(field.form);
+                        if (field && this.field.form) {
```

Comment:
We should whenever we plan to refactor the engaging of the visibility check. At the moment it is still as it was cause it was still not done yet .